src/MSAL.PCL/Internal/Http/HttpClientFactory.cs

```diff
@@ -35,7 +35,17 @@ internal class HttpClientFactory
         private static HttpClient _client;
         private static readonly object LockObj = new object();
         public static bool ReturnHttpClientForMocks { set; get; }
+        public const long MaxResponseContentBufferSizeInBytes = 1024*1024;
 
+        private static HttpClient CreateHttpClient()
```

Comment:
While the implementation in this PR is functionally correct, I would recommend to change it slightly.

    // The newly implemented CreateHttpClient() can be avoided, if we refactor the GetHttpClient() into something like this
    public static HttpClient GetHttpClient()
    {
        var httpClient = new HttpClient(HttpMessageHandlerFactory.GetMessageHandler(ReturnHttpClientForMocks))
             {
                 MaxResponseContentBufferSize = MaxResponseContentBufferSizeInBytes
             };
        if (ReturnHttpClientForMocks) {
            return httpClient;
        }

        if (_client == null)
        {
            lock (LockObj)
            {
                if (_client == null)
                {
                    _client = httpClient;
                }
            }
        }
    }

The reason for the change: At a hindsight, the preexisting GetHttpClient() contains duplicated code already. The implementation in this PR consolidates them into one, which is good, with the help of a new shared helper function. But we could avoid such extra helper function, if we don't have duplication logic in the first place. Also, aesthetically speaking, current implementation in the PR ends up with 2 static method with similar names, `GetHttpClient()` and `CreateHttpClient()`. Future caller/maintainer may not intuitively notice one of them are private.

Last but not the least, the above implementation minimizes the codes would be run within a lock.